nes/CMakeLists.txt

```diff
@@ -26,8 +26,9 @@ set_target_properties(${PROJECT_NAME}
 )
 
 target_link_libraries(${PROJECT_NAME}
-    PRIVATE
+    PUBLIC
```

Comment:
Maybe forward-declare the core components and leave this private? Then it's up to the consumer of the n_e_s::nes library if they want to pull in n_e_s::core to access e.g. the mmu.

Reply:
Good idea, but we can't do it right now. Not possible to forward declare nested classes (ICpu::Registers). We need to split those classes then.
There is a paper (P0289R0) to add support for this into the standard, but not sure about the status. It is not in c++20 at least.

Reply:
I can make Registers its own class if you want :)

Reply:
Rip. I'd be happy to move that class out of ICPu to keep Core and Nes separate. :D

Reply:
Done :D